src/components/comment/comment.ts

```diff
@@ -0,0 +1,41 @@
+export const initComments = () => {
+  const commentsWithRepliesEnabled = document.querySelectorAll(
+    '.js-comment-with-replies-enabled'
+  );
+
+  for (const comment of [...commentsWithRepliesEnabled]) {
+    const replyButton = comment.querySelector('.js-comment-reply-button');
+    const replyForm = comment.querySelector('.js-comment-reply-form');
+    // The cancel button is inside the Comment Form, so we need to reach inside
+    // that component to grab the button. This feels a little awkward since
+    // we're directly manipulating a child component, but this is a lot simpler
+    // than other potential solutions. Other options included:
+    // - Having the Comment Form emit a `cancel` event
+    // - Having the Comment Form have a block that you could pass a button into
+    const cancelButton = comment.querySelector('.js-cancel-reply');
+
+    replyButton?.addEventListener('click', () => {
+      comment.classList.add('is-replying');
+      replyButton.setAttribute('hidden', '');
+      replyForm?.removeAttribute('hidden');
+      const firstInput = replyForm?.querySelector(
+        'textarea, input'
+      ) as HTMLElement | null;
```

Comment:
Since we are in a .ts file we can use a type parameter. querySelector accepts a type parameter to tell it what types of element(s) it may find: https://www.typescriptlang.org/play?#code/MYewdgzgLgBATgUwA4BsCeAxEcC2MC8MA9AFQwCWA5mNgjDnSUTAAwwCGEHYaHXAEgBUAsgBkAoigQMwUAFBzQkWADNycaAEkwSAK6xCiVJmw4A-ADoAjroRw0AZQRTgUbAB4hYwQgAeUAEFEdklpBFkYAB8YL1FtPShQmSgAPgAKAHIoPyh2YIAaCh19DIBKIA

It'll automatically add the `| null` when you are using the type parameter

Reply:
Oh neat, thanks! I'll try that out.